cmd/clusterctl/client/config/reader_viper.go

```diff
@@ -61,8 +61,11 @@ func (v *viperReader) Init(path string) error {
 	viper.AllowEmptyEnv(true)
 	viper.AutomaticEnv()
 
-	if err := viper.ReadInConfig(); err != nil {
-		return err
+	err := viper.ReadInConfig()
+	if err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+			return err
+		}
```

Comment:
I did it this way because I still want to return errors from `viper.ReadInConfig()`. There may be other errors regarding parsing the config or even if viper supports the config extension.
Ideally, I could've done a  `os.Stat(viper.ConfigFileUsed())` but I believe there is a behavioral bug for this since it could return `""`.

Reply:
i think a comment on top will be useful for future readers.